posthog/auth.py

```diff
@@ -111,7 +111,13 @@ def find_key_with_source(
         if "HTTP_AUTHORIZATION" in request.META:
             authorization_match = re.match(rf"^{cls.keyword}\s+(\S.+)$", request.META["HTTP_AUTHORIZATION"])
             if authorization_match:
-                return authorization_match.group(1).strip(), "Authorization header"
+                token = authorization_match.group(1).strip()
+
+                if token.startswith(
+                    "pha_"
+                ):  # This is an OAuth access token, not a personal API key. We assume all other tokens are personal API keys, since legacy tokens may not have been prefixed.
```

Comment:
It's slightly ambiguous what "legacy tokens" is referring to. I'm assuming it means legacy personal api keys (as opposed to legacy oauth access tokens?)

```suggestion
                ):  # This is an OAuth access token, not a personal API key. We assume all other tokens are personal API keys, since legacy personal api keys may not have been prefixed.
```